crates/rspack_error/tests/fixtures/should_not_warn_export_destructring/a.js

```diff
@@ -0,0 +1,8 @@
+const foo = {
+  noop: () => {},
+	rename: 114,
+	otherFields: 514,
+};
+
+export const { noop, rename: reassign, ...rest } = foo;
```

Comment:
I'm not sure it works. 

Even though, we could analyze which variable is not used in `{ noop, rename: reassign, ...rest } = foo`.

But the tree shaker(which is the SWC minfier) we rely on doesn't shake unused variables in pattern like `{ noop, rename: reassign, ...rest } = foo`.

See https://play.swc.rs/?version=1.3.44&code=H4sIAAAAAAAAA0vOzysuUUjLz1ewVajmUlDIy88vsFLQ0FSwtVOortXh4ixKzUvMTbVSMDQ0AfLySzJSi9wyU3NSiq0UTEFCtdZcyWAzqsF6dRQgGnQUkJQq1AJNB9oBVpmfk6qXk5%2BuAVKtCQCVeh%2FZfwAAAA%3D%3D&config=H4sIAAAAAAAAA31VuY7jMAzt5ysC1VssUmwx7Q7S7TcIikQ5nLVFQ6QyCQb596GP3HK6WI%2BPN1%2B%2B31Yr88nevK%2B%2B9ad%2B9C4z5Mu3vvAxiTvoi5FjD%2Bwz9mJ%2BndEAnrITyqwWkgtcEOHD%2FDS%2BnCbASHaJI%2BXuNkZh%2BICICTaU%2F7aOeYPQhieXGZyXG97wVJJgB0N6rgh1TtCbGT7dB3a5gYFtgNe%2F1%2Bu5BtMS8cCPruU5lOkwYTzeZuip6zMw30dXl6WDJHzPn7FMX48l6PuWqAXtwTJiHVtMAg3kmmNPbet6Brt3T12fM3UZmWohBrAIBNtn6qt4CihISWM%2BowFcsJ4CVCDM4AX3UKNpLKUl1vIq9YxwgG1pmnHzHtiwd21xUokJh3Ekmm3F646QxcaSai2cwIUeTODc3EcmRptBSk7PvE%2FCtDCT%2FwDagWGxk%2Bug5ne0iLpPS%2Bz4kolJrwflWMF1v2tVJmi0qRYxVjo7dAayYG2aGULxMHTW19KZ4YX2MQawEKPuSsU1f6H4XS3oID0UK4DO18XaVk2AvVzhAj4cxAt4o1VKfcFmC1Wc3TLKx25L7YsAHciOwgsDHYXQMpxVJQ79Ml5SGIU1VE1Ud0NVBPQAhGw76uXTbuh5qEfbtLS9ysRZcy9i3bnUXO99kuK32cB0FMoIzn87w3wnaf5jrkZnFb4kbpD%2FnZlj0NMPG2ZSgsIGAAA%3D

Reply:
The visitor change here only involves export symbols. Is swc-minifier's optimizing behavior affecting the visitor? 

What if our analysis has different output with swc-minifiers?

Reply:
The process is

1. We analyze which variable is unused.
2. Remove the `export` keyword for unused variables
3. perform SWC minifier to shake unused variables/AST nodes

The problem is we successfully remove the `export` keyword for unused variables but the SWC minifier doesn't support to shake code pattern mentioned above.

Reply:
Gotcha. Where is the remove export keyword code? The problem here is that we cannot remove export for cases like this 

```
export const { used, notUsed } = {...}
```

This is a partial used binding, changing it to `const { ...} = {...}` is not correct.

Reply:
Yeah. That's the problem for patterns like `export const { used, notUsed } = {...}`. 

I have investigated such cases before. That's why I'm not sure this PR works.

Reply:
Indeed tree shaking will not work on export destructring.

![image](https://user-images.githubusercontent.com/2883231/229016463-64aa2107-5e38-4ecc-abf4-bf87eba37eea.png)
![image](https://user-images.githubusercontent.com/2883231/229016477-92845ad6-d8c5-4d1c-9d29-386630a09d55.png)

Reply:
If you have an interest in tree-shaking, you could try to investigate this case.

@IWANABETHATGUY Have we supported tree-shaking on cases like this?

```js
// a.js
export const unused = 1, used = 1;

// index.js
import { used } from './a.js'
console.log(used)
```

=> transform

```js
// a.js
const unused = 1;
export const used = 1;

// index.js
import { used } from './a.js'
console.log(used)
```

=> finally

```js
// a.js
export const used = 1;

// index.js
import { used } from './a.js'
console.log(used)
```

Reply:
yes, we have support this case